login/admin.py

```diff
@@ -0,0 +1,7 @@
+from django.contrib import admin
+
+# Register your models here.
+
+from .models import CofkUser
+
+admin.site.register(CofkUser)
```

Comment:
I suggest to add forms.py:

(this is borrowed from: https://docs.djangoproject.com/en/4.0/topics/auth/customizing/#a-full-example)


```python
from django import forms
from django.core.exceptions import ValidationError

from login.models import CofkUser


class CofkUserForm(forms.ModelForm):
    password1 = forms.CharField(label='Password', widget=forms.PasswordInput)
    password2 = forms.CharField(label='Password confirmation', widget=forms.PasswordInput)

    def clean_password2(self):
        # Check that the two password entries match
        password1 = self.cleaned_data.get("password1")
        password2 = self.cleaned_data.get("password2")
        if password1 and password2 and password1 != password2:
            raise ValidationError("Passwords don't match")
        return password2

    def save(self, commit=True):
        # Save the provided password in hashed format
        user = super().save(commit=False)
        user.set_password(self.cleaned_data["password1"])
        if commit:
            user.save()
        return user

    class Meta:
        model = CofkUser
        exclude = ('password', )

```

and change admin.py to:

```python

class CofkUserAdmin(admin.ModelAdmin):
    form = CofkUserForm


admin.site.register(CofkUser, CofkUserAdmin)
```

this will enable user creation with passwords via the admin interface rather than CLI

Reply:
as discuss in stack, we not sure that will client create User by `django admin page`, we might need to create User page as `Location` section.
If client agree to use `django admin page`, we can apply your suggested change in the other branch.

Reply:
I don't think it's a good idea to create general users as super users.

Regardless of whether or not the client wants to use the admin interface (which I don't think they will), it is a useful interface for us to make changes. Say if we need to change some property of a user. The admin interface is also useful for browsing users.

I think we should implement these changes for us, as technical admins, to be able to change things.

Reply:
we should discuss this with Client. 
we need to let them know we will keep admin page as technical admins
otherwise, If they don't want a `django admin page` but we keep the page, It will become backdoor for us.
@anusharanganathan , any comment about admin page?

Reply:
@philipkcl even if we are not technical admin, the project has a technical administrator who would want overall access, so it is good to keep django admin page. John Pybus is their technical administrator. The only question to ask the client is, is this page too complicated for the EMLO administrator who is not very technical. If so, we can create a simpler dashboard, but for now let's stick with the Django admin pages. .

Reply:
At the most EMLO will have three super users (Cottage Labs, EMLO technical admin, EMLO admin). 2 or 3 more users will have editor access and everyone else will only be given read access. They are very careful about who can be super user.

Reply:
> At the most EMLO will have three super users (Cottage Labs, EMLO technical admin, EMLO admin). 2 or 3 more users will have editor access and everyone else will only be given read access. They are very careful about who can be super user.

For this we will use user [groups](https://docs.djangoproject.com/en/4.0/topics/auth/default/#groups). It is also good to manipulate via the Django admin UI.

Reply:
> The only question to ask the client is, is this page too complicated for the EMLO administrator who is not very technical.
* `django admin page` is simple Content management system for Admin editing record.
* it is easy to use.
* It support customize but not sure is that can be handle Client's future requirement.
* ref: https://docs.djangoproject.com/en/4.0/intro/tutorial07/

Since, we want to keep admin page, I will create new branch for fixing the admin page.